students/dominic_martin/lesson05/mailroom3.py

```diff
@@ -1,97 +1,58 @@
-	global donations
-	print()
-	newDonation = int(input("Enter a donation amount:"))
-	enumerate(donations.append[nameIdx](newDonation))
-	print()
-	print(str(names[nameIdx]) + " has donated " + "$" + str(newDonation))
-	tyemail(nameIdx)
-
-
-def tyemail(nameIdx):
-	global names
-	print()
-	print(str(names[-1]) + ", thank you so much for your donation.  It's going to a wonderful cause to help lessen the greenhouse emissions from our production plant.  Thanks again and Happy Holidays!")
-	
-
-def genReport():
-	global namesDonations
-	print()
-	width1 = 27
-	p = ('Name:', 'Amount:')
-	print(f'{p[0]:{width1}}{p[1]:{width1}}')
-	# for name, donation in donation_list1.items():
-	for name, donation in namesDonations.items(): # new
-		print(f'{name:{width1}}{"$"+str(donation):{width1}}')
-			
-
-def writeLetters():
-	global namesDonations
-	for key in namesDonations.keys():
-		with open('%s.txt'%(key,), 'w') as f:
-			f.write(key  + "\n\nThanks for your donation!\n\nAll the Best,\n\nThe Mailroom Team") 		
-		
-
-def quitAll():
-	print()
-	print("Thanks for using The Mailroom Application!  Goodbye!")
-
-
-while True:
-	try:
-		a = int(input("What would you like to do?\n(1) - Send a Thank You?\n(2) - Create a Report?\n(3) - Send Letters to Everyone?\n(4) - Quit?\nEnter your response:"))
-		if a in range(0,4):
-			choices(a)
-	except TypeError:
-	
-		print("\nIncorrect Input.  Try again.")
-	else:
-		break
-		
-
-	
-
-
+names_donations = {name: donation for name, donation in zip(names, donations)}   
+ 
+
+def send_thanks():
+    """Print a thank you message to the selected person."""
+    while True:
+        try:
+            choose_response = int(input(pm.choose_prompt))
+            print("\n" + names[choose_response - 1] + ", thanks for your donation." +"\n")
+        except:
+            print("Try again with a number between 1 and 5.")
+            continue
+        else:
+            break
+
+def thanks_everyone():
+    """Open and write a .txt file with a thank you message for every person."""
+    for key in names_donations.keys():
+        with open('%s.txt'%(key,), 'w') as f:
+            f.write(key  + "\n\nThanks for your donation!\n\nAll the Best,\n\nThe Mailroom Team") 
+
+def gen_report():
+    """Generate a report containing all names and donations."""
+    print()
+    width1 = 27
+    p = ('Name:', 'Amount:')
+    print(f'{p[0]:{width1}}{p[1]:{width1}}')
+    for name, donation in names_donations.items():
+        print(f'{name:{width1}}{"$"+str(donation):{width1}}')
+
+def exit_program():
+    """Exit the program."""
+    print("\nThanks for using The Mailroom Application!  Goodbye!")
+    sys.exit()
+
+def main():
+    """Prompt user to make a choice at the start of program."""
+    while True:
+        try:
+            response = input(pm.prompt)
+            switch_func_dict = {
+            "1": send_thanks,
+            "2": thanks_everyone,
+            "3": gen_report,
+            "4": exit_program,
+            }
```

Comment:
`switch_func_dict` needs to go outside of the loop, it does not need to be redefined each time